local_planner/src/local_footstep_planner.cpp

```diff
@@ -174,47 +171,67 @@ void LocalFootstepPlanner::computeFootPlan(
             body_vel_touchdown, ref_body_vel_touchdown, body_ang_vel_touchdown,
             ref_body_ang_vel_touchdown, grf_midstance;
 
-        // Compute the horizon index of midstance
-        int midstance = i + half_duty_cycle;
-
-        // Compute body pose at midstance from either the trajectory or Raibert
-        // heuristic directly
-        if (midstance < horizon_length_) {
-          body_pos_midstance = body_plan.block<1, 3>(midstance, 0);
-          body_rpy_midstance = body_plan.block<1, 3>(midstance, 3);
-          // grf_midstance = grf_plan.block<1, 3>(midstance, 3 * j);
+        // Compute the number of timesteps corresponding to the end of stance
+        // phase
+        int end_of_stance = i + period_ * duty_cycles_[j];
```

Comment:
We probably want this to use the contact schedule rather than the period/duty cycles, similar to the already existing getNextContactState.